File: src/main/java/dev/discordmk/neutron/ProjectFrame.java
package dev.discordmk.neutron;

import dev.discordmk.neutron.components.NeutronMenuBar;
import dev.discordmk.neutron.panels.BottomPanel;
import dev.discordmk.neutron.panels.WorkspacePanel;
import dev.discordmk.neutron.util.AssetGetter;

import javax.swing.*;
import java.awt.*;

public class ProjectFrame extends JFrame {
    public BottomPanel bottomPanel;

    public ProjectFrame() {
        super("DiscordMK");
        setDefaultCloseOperation(EXIT_ON_CLOSE);
        setSize(800, 600);
        setLocationRelativeTo(null);
        setJMenuBar(new NeutronMenuBar());
        setLayout(new BorderLayout());

        bottomPanel = new BottomPanel();

        add(new WorkspacePanel(this), BorderLayout.CENTER);
        add(bottomPanel, BorderLayout.SOUTH);

        ImageIcon icon = AssetGetter.getImageIcon("icon");
        if (icon != null) setIconImage(icon.getImage());
        setVisible(true);
    }
}

File: src/main/kotlin/dev/tonimatas/discordmk/frames/ProjectFrame.kt
package dev.tonimatas.discordmk.frames

import dev.tonimatas.discordmk.components.NeutronMenuBar
import dev.tonimatas.discordmk.panels.BottomPanel
import dev.tonimatas.discordmk.panels.WorkspacePanel
import dev.tonimatas.discordmk.util.AssetGetter
import java.awt.BorderLayout
import javax.swing.JFrame

class ProjectFrame : JFrame("DiscordMK") {
    var bottomPanel: BottomPanel

    init {
        defaultCloseOperation = EXIT_ON_CLOSE
        setSize(800, 600)
        setLocationRelativeTo(null)
        jMenuBar = NeutronMenuBar()
        layout = BorderLayout()

        bottomPanel = BottomPanel()

        add(WorkspacePanel(this), BorderLayout.CENTER)
        add(bottomPanel, BorderLayout.SOUTH)

        val icon = AssetGetter.getImageIcon("icon")
        iconImage = icon.image
        isVisible = true
    }
}